src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -333,6 +333,10 @@ private static byte[] EncodeStringName()
         {
             var type = typeof(string);
             var name = type.AssemblyQualifiedName;
+            
+            // ResourceWriter always write the resources with legacy core library name (i.e. mscorlib).
+            // Ensure that when we read the type name, we update the library name to be the same.
+            name = name.Replace("System.Private.CoreLib", "mscorlib");
```

Comment:
We should just hardcode the string with mscorlib name instead of asking for name first via reflection and then doing find and replace because of we are not happy with what we got. Similar to what resourcewriter does. I will be a bit more efficient as well.